spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/KStreamBoundElementFactory.java

```diff
@@ -52,8 +53,12 @@
 
 	@Override
 	public KStream createInput(String name) {
-		ConsumerProperties consumerProperties = this.bindingServiceProperties
-				.getConsumerProperties(name);
+		BindingProperties bindingProperties = this.bindingServiceProperties.getBindingProperties(name);
+		ConsumerProperties consumerProperties = bindingProperties.getConsumer();
+		if (consumerProperties == null) {
+			consumerProperties = this.bindingServiceProperties.getConsumerProperties(name);
+			consumerProperties.setUseNativeDecoding(true);
+		}
```

Comment:
This code here appears to be identical to the one in `GlobalKTableBoundElementFactory.createInput(..)` and `KTableBoundElementFactory.createInput(..)`. Have you considered some abstract class?

Reply:
sure, I am going to refactor a bunch of these classes. Can I address these concerns as part of that?